http-server/src/main/java/org/triplea/server/error/reporting/ErrorReportController.java

```diff
@@ -1,42 +1,48 @@
 package org.triplea.server.error.reporting;
 
 import java.util.function.Function;
+import java.util.function.Predicate;
 
+import javax.annotation.Nonnull;
 import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.GET;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.core.Context;
 import javax.ws.rs.core.MediaType;
 
+import org.triplea.http.client.error.report.ErrorUploadClient;
 import org.triplea.http.client.error.report.ErrorUploadRequest;
 import org.triplea.http.client.error.report.ErrorUploadResponse;
 
-import lombok.AllArgsConstructor;
+import lombok.Builder;
 
 /**
  * Http controller that binds the error upload endpoint with the error report upload handler.
  */
-@AllArgsConstructor
-@Path(ErrorReportController.ERROR_REPORT_PATH)
+@Builder
 @Produces(MediaType.APPLICATION_JSON)
+@Path("/")
 public class ErrorReportController {
 
-  static final String ERROR_REPORT_PATH = "/error-report";
-
+  @Nonnull
   private final Function<ErrorReportRequest, ErrorUploadResponse> errorReportIngestion;
+  @Nonnull
+  private final Predicate<String> errorReportRateChecker;
 
   @POST
+  @Path(ErrorUploadClient.ERROR_REPORT_PATH)
```

Comment:
What's the difference between using a `@<REQUEST_TYPE>` + `@Path` annotation vs `@RequestLine`?
The latter one was used earlier in this diff.

Reply:
Http client vs Server. RequestLine is Feign,  `@Path` is server side (drop-wizard):
- import feign.RequestLine;
- import javax.ws.rs.Path;

Reply:
Ah, my bad. Thanks for the clarification